src/rocm_docs/core.py

```diff
@@ -200,45 +213,95 @@ def _set_all_article_info(
-            font_awesome_os += '<i class="fa-brands fa-windows fa-2xl"></i>'
+            font_awesome_os += '<i class="fa-brands fa-windows fa-2xl fa-fw"></i>'
+
+        page_key = Path(page).stem
+        if page_key in source_map.keys():
+            modified_path = source_map[Path(page).stem]
+        else:
+            modified_path = page
 
-        modified_info = article_info.replace("<!--fontawesome-->", font_awesome_os)
+        modified_info = article_info.replace("<!--osicons-->", font_awesome_os)
         modified_info = modified_info.replace("AMD", app.config.all_article_info_author)
-        modified_info = modified_info.replace("2023", app.config.all_article_info_date)
-        modified_info = modified_info.replace("5 min read", app.config.all_article_info_read_time)
+        modified_info = modified_info.replace("2023", _get_time_last_modified(modified_path))
+        modified_info = modified_info.replace("5 min read", _estimate_read_time(page))
         
         _write_article_info(page, modified_info)
 
 
-def _get_all_pages(output_directory: str) -> List[str]:
-    all_pages = []
+def _get_all_pages(output_directory: str):
+    html_pages = list()
+    source_map = dict()
+
     for root, _, files in os.walk(output_directory):
         for file in files:
             if file.endswith(".html"):
-                all_pages.append(os.path.join(root, file))
-    return all_pages
+                html_pages.append(os.path.join(root, file))
+
+    for root, _, files in os.walk("."):
+        for file in files:
+            if file.endswith(".rst") or file.endswith(".md"):
+                file_key = Path(file).stem
+                source_map[file_key] = os.path.join(root, file)
+    
+    return (html_pages, source_map)
+
+
+def _get_time_last_modified(path: str) -> str:
+    last_time_modified = os.path.getmtime(path)
+    global_time = time.gmtime(last_time_modified)
+    date_info = time.strftime("%B %d, %Y", global_time)
+    return date_info
+
+
+def _estimate_read_time(file_name: str) -> str:
+    def is_visible(element):
+        if element.parent.name in ['style', 'script', '[document]', 'head', 'title']:
+            return False
+        elif isinstance(element, bs4.element.Comment):
+            return False
+        elif element.string == "\n":
+            return False
+        return True
+    
+    def count_words(text, avg_word_len):
+        words = 0
+        for line in text:
+            words += len(line)/avg_word_len
+        return words
+
+    WORDS_PER_MIN = 200
+    AVG_WORD_LEN = 5
+
+    file = open(file_name, "r")
+    html = file.read()
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+    page_text = soup.findAll(text=True)
+    visible_page_text = filter(is_visible, page_text)
+    average_word_count = count_words(visible_page_text, AVG_WORD_LEN)
+    time_minutes = max(1, int(average_word_count // WORDS_PER_MIN))
+    return f"{time_minutes} min read time"
```

Comment:
Are we sure we **want** to do this? In my view it would be better to not show the reading time, when no human has given attention to it, instead of showing a crude estimate based on word count.

Reply:
Let's keep it for now.